### ISSUE TITLE
add negative test of nameless/non-method/remote-function call

### ISSUE THREAD

**Issue by warner** [Agoric/Agoric#4737]
## What is the Problem Being Solved?

https://github.com/Agoric/agoric-sdk/issues/61#issuecomment-1059459378 points out that swingset does not currently support cross-vat "nameless calls", like `E(obj)(args)` (intended to be used with `obj = Far('iface', args => blah)`). The exact failure mode is a little obscure: it rejects the result promise, but the error is triggered by `insistMessage` when it sees `message.method` is not a string (it is `undefined`).

We have a plan to support this, and it would probably be useful, but we haven't gotten around to it yet. We don't have a test that demonstrates that it fails (or what the failure looks like), nor do we have any documentation recommending for or against using the pattern.

The tricky part is that `HandledPromise` *does* have support for this, so it works locally, just not when crossing a vat boundary. That's not good: code which works in some configurations would fail in others.

We don't have time to implement #61 now, but it might be good to:
* move the assertion up into liveslot's `applyMethod` handler, and include the phrase "`E(obj)(args)`" in the error so users know what to look for
  * (the handler is called in a later turn, so we can't give them a stack trace to where they strayed beyond the implementation's abilities, and there's no method name to share with them, but describing the troublesome pattern might still help)
* write a unit test that attempts the call and asserts that the result promise is rejected, with the expected error message
* add a `console.log()` complaint when this happens too
  * in my test, where I didn't attach an error handler to the `p = E(obj)(args)`, the error seems to have been silently discarded
* and add a pointer to #61 so users who hit this can learn more

@michaelfig @erights do you think this limitation is serious enough to consider looking for a way to disable nameless *local* calls? I don't know how we would, but maybe we should look?

**Comment by michaelfig** [comment]
> @michaelfig @erights do you think this limitation is serious enough to consider looking for a way to disable nameless _local_ calls? I don't know how we would, but maybe we should look?

No.  The semantics of `E(x)...` and `E.get(x)...` are that you need to handle rejected promises, or chain these calls together so that the rejected promise is propagated.  Dropping them on the floor is the caller's fault, because they can fail for arbitrary reasons.

The `unhandledPromiseRejection` callback should have caught this error.  I'm confused why it didn't.

**Comment by michaelfig** [comment]
> The `unhandledPromiseRejection` callback should have caught this error. I'm confused why it didn't.

OIC, when running in XS we don't get such a callback.  ISTR @erights or @kriskowal had a sketch for what ideal unhandled promise rejection tracking would look like.  Can we suggest that to the XS folk?

**Comment by michaelfig** [comment]
> OIC, when running in XS we don't get such a callback.

Here is the code in `packages/xsnap/xsnap-native/xsnap/sources/xsnapPlatform.c`:

```c
	case XS_UNHANDLED_EXCEPTION_EXIT:
		fxReport(the, "unhandled exception: %s\n", fxToString(the, &mxException));
		mxException = mxUndefined;
		break;
	case XS_UNHANDLED_REJECTION_EXIT:
		fxReport(the, "unhandled rejection: %s\n", fxToString(the, &mxException));
		mxException = mxUndefined;
		break;
```

Both unhandled rejections and unhandled exceptions are swallowed up (`fxReport` is a no-op).  We should probably ensure they are reflected to the host as a console message, but that requires some rewiring we should do at an opportune time.

Also, I found unhandled rejections have an `mxException` of `undefined` before this switch statement, so they're hard to diagnose.  I reported that to the XS folks.

**Comment by Tartuffo** [comment]
Liveslots handler can reject this, moves the error from the kernel closer to the user.  Local calls will still work.
